charts/etcd/templates/etcd-statefulset.yaml

```
@@ -53,12 +53,17 @@ spec:
         command:
         - /var/etcd/bin/bootstrap.sh
         readinessProbe:
-          httpGet:
+          exec:
+            command:
+            - /usr/bin/curl
 {{- if .Values.etcd.enableTLS }}
-            scheme: HTTPS
-{{- end }}
-            path: /healthz
-            port: {{ .Values.backup.port }}
+            - --key=/var/etcd/ssl/client/tls.key
+            - --cacert=/var/etcd/ssl/ca/ca.crt
+            - --cert=/var/etcd/ssl/client/tls.crt
+            - https://0.0.0.0:{{ .Values.etcd.clientPort }}/health
```

```suggestion
            - https://{{ .Values.name }}-local:{{ .Values.etcd.clientPort }}/health
```
Otherwise `0.0.0.0` must be part of the server cert.

---

Changed it. We need to modify both client and peer certificates anyway. Earlier, client certs were not considering multinode scenario